plugins/modules/fabric_virtual_networks_workflow_manager.py

```diff
@@ -0,0 +1,3480 @@
+                the class state with the results of the operations (created, updated, or no action needed).
+        Description:
+            This function processes a list of virtual networks and performs necessary operations to either create or update
+            the configurations in Cisco Catalyst Center. The steps are as follows:
+            - Iterates over the `virtual_networks` list, extracting each VN's name (`vn_name`).
+            - Checks if the given virtual network already exists in the Cisco Catalyst Center:
+                - If it exists, the function retrieves the current VN configuration and checks whether an update is required.
+                If so, it adds the update payload to the collection. If no update is necessary, it logs a message indicating
+                no action is needed.
+                - If the virtual network does not exist, the function prepares a payload for creating the virtual network
+                and adds it to the creation payload collection.
+            - After collecting the payloads, the function proceeds to:
+                - Create any new virtual networks that were not found in the existing configuration.
+                - Update any virtual networks that require changes.
+            - The function logs the outcome for each virtual network and ensures that the operation status is reflected
+            accordingly.
+            In the event of errors or failures during the process, the function logs the issue and updates the operation
+            result accordingly.
+        """
+
+        add_vn_payloads, update_vn_payloads = [], []
+        for vn_details in virtual_networks:
+            vn_name = vn_details.get("vn_name")
+            vn_payload = {"virtualNetworkName": vn_name}
+
+            if self.have.get("l3_vn_name") and vn_name in self.have.get("l3_vn_name"):
+                # Given VN already present in Cisco Catalyst Center, check vn needs update or not.
+                vn_in_ccc = self.get_vn_details_from_ccc(vn_name)
+                vn_needs_update = self.is_vn_needs_update(vn_details, vn_in_ccc)
+                if vn_needs_update:
+                    self.updated_virtual_networks.append(vn_name)
+                    update_vn_payloads.append(self.update_payload_vn(vn_details, vn_in_ccc))
+                else:
+                    # Given Virtual network doesnot need any update
+                    self.no_update_virtual_networks.append(vn_name)
+                    self.msg = "Given Virtual network '{0}' does not need any update".format(vn_name)
+                    self.log(self.msg, "INFO")
+                    self.result["response"] = self.msg
+            else:
+                self.created_virtual_networks.append(vn_name)
+                vn_payload = self.create_vn_payload(vn_details)
+                add_vn_payloads.append(vn_payload)
```

Comment:
self.log("Virtual Network '{0}' is new and will be created.".format(vn_name), "INFO")